src/LiveObjects.cpp

```diff
@@ -658,10 +777,21 @@ void LiveObjectsGSM::messageCallback(int msg)
 
 void LiveObjectsGSM::addNetworkInfo()
 {
- JsonObject obj = easyDataPayload[JSONVALUE].createNestedObject("networkInfo");
- obj["connection_status"] = m_GSMAcces.status() == GSM_READY ? "connected":"disconnected";
- obj["strength"] = m_GSMScanner.getSignalStrength();
- obj["carrier"]=m_GSMScanner.getCurrentCarrier();
+  String strength=m_GSMScanner.getSignalStrength();
+  String carrier = m_GSMScanner.getCurrentCarrier();
+  if(m_Protocol == SMS)
+  {
+    addToStringPayload(m_GSMAcces.status() == GSM3_NetworkStatus_t::GSM_READY);
+    addToStringPayload(strength);
+    addToStringPayload(carrier);
+  }
+  else
+  {
+    JsonObject obj = easyDataPayload[JSONVALUE].createNestedObject("networkInfo");
+    obj["connection_status"] = m_GSMAcces.status() == GSM_READY ? "connected":"disconnected";
+    obj["strength"] = m_GSMScanner.getSignalStrength();
+    obj["carrier"]=m_GSMScanner.getCurrentCarrier();

```

Comment:
to consider creating a universal function for adding data to payload for every protocol(mqtt, sms, lora in future)